CRM/Utils/Hook.php

```diff
@@ -1635,6 +1635,29 @@ public static function disable() {
     );
   }
 
+  /**
+   * Alter redirect.
+   *
+   * This hook is called when the browser is being re-directed and allows the url
+   * to be altered.
+   *
+   * @param \Psr\Http\Message\UriInterface $url
+   * @param string $outputFormat
+   *   The output format for the request. Generally this will be NULL or json.
+   *   If json it will lead to json output being returned rather than a true redirect.
+   * @param array $context
+   *   Optional additional information about context.
+   *
+   * @return null
+   *   the return value is ignored
+   */
+  public static function alterRedirect(&$url, &$outputFormat, $context) {
```

Comment:
Since `$url` is now an object, the `&` operator isn't necessary anymore

Reply:
Good point - I've done that. I'm a bit torn though now as technically I should re-order & pass $outputFormat first since it's still a reference - but it is also the 'barely worth including' param in the set - I only passed it since @agh1 seemed to see some possible use case

Reply:
Is there a rule that says that references should come before the other params? Does the order make any difference in that case?

Reply:
IMHO, there's no obligation to put references first. Rather, just put the most important/salient field first.

In the case of alter-style hooks, the most important/salient field is also often a reference, so it may often look like references come first. But that's happenstance (to my eye).